docs/extensibility/index.md

```diff
@@ -21,6 +22,41 @@ The strategy options contain properties of following types:
 - **Delegates**: For example when a strategy needs to raise an event, or generate a value. In general, the delegates should by asynchronous.
 - **Arguments**: Used by the delegates to pass the information to their consumers.
 
+### Component diagram
+
+This diagram depicts how the built-in types interact with the custom built types (with green border)
```

Comment:
I'm not sure about the use of green here - the contrast doesn't seem great when you don't use dark mode:

![image](https://github.com/App-vNext/Polly/assets/1439341/1028d8aa-5352-4f37-a013-fe6ae8e62811)

Reply:
It seems to me that only **red** and **orange** are somewhat visible in light mode.
<img width="600" alt="Screenshot 2023-12-19 at 12 18 44" src="https://github.com/App-vNext/Polly/assets/57183693/d0bbf724-6fc6-48e8-ace1-477088a77c5c">

<img width="691" alt="Screenshot 2023-12-19 at 12 18 28" src="https://github.com/App-vNext/Polly/assets/57183693/c1ba8a8f-9809-4886-8142-658126dd084d">

Which one do you prefer? Or do you have other suggestion?

Reply:
Or alternatively we can change the [shape of the nodes](https://mermaid.js.org/syntax/flowchart.html#node-shapes).

Reply:
I like the idea of changing the shapes as then that works for people who may have colour-blindness.